In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
from sklearn import svm, datasets
from matplotlib.colors import ListedColormap

In [2]:
iris = datasets.load_iris()
X = iris.data[:, [0,1,2,3]]
y = iris.target

results = []

In [3]:
_svm = svm.SVC(kernel='linear')
_svm.fit(X,y)
result = _svm.score(X,y)
results.append(('linear',round(result,2)))

In [4]:
_svm = svm.SVC(kernel='poly')
_svm.fit(X,y)
result = _svm.score(X,y)
results.append(('poly',round(result,2)))

In [5]:
_svm = svm.SVC(kernel='sigmoid')
_svm.fit(X,y)
result = _svm.score(X,y)
results.append(('sigmoid',round(result,2)))

In [6]:
_svm = svm.SVC(kernel='rbf')
_svm.fit(X,y)
result = _svm.score(X,y)
results.append(('rbf',round(result,2)))

In [7]:
from prettytable import PrettyTable
table = PrettyTable(['Kernel','Accuracy'])
for i in results:
    table.add_row(i)
print(table)

+---------+----------+
|  Kernel | Accuracy |
+---------+----------+
|  linear |   0.99   |
|   poly  |   0.97   |
| sigmoid |   0.07   |
|   rbf   |   0.97   |
+---------+----------+


# Discussion

In this part of the assignment, we were tasked to use the SVM classifier on the iris data set using all the features and catagories at once. The catagories being virginica, versicolor, and setora and the four features being sepal length, sepal width, petal length, petal width. The classifier was ran 4 times, each time using a different kernel. As seen above, the ones used were linear, poly, sigmoid, and rbf with linear having the best accuracy at a score of 0.99 and sigmoid having the worst accuracy by far at a score of 0.07. Poly and rbf were both at 0.97, making them slightly worst then linear.